bin/ruby-build

```diff
@@ -244,24 +244,39 @@ compute_md5() {
   fi
 }
 
+has_checksum_support() {
+  local checksum_command="$1"
+  local has_checksum_var="HAS_CHECKSUM_SUPPORT_${checksum_command}"
+
+  if [ -z "${!has_checksum_var+defined}" ]; then
+    printf -v "${has_checksum_var}" '%s' $(echo test | "$checksum_command" >/dev/null; echo $?)
+  fi
+  return "${!has_checksum_var}"
+}
+
 verify_checksum() {
-  # If there's no SHA2 support, return success
-  [ -n "$HAS_SHA2_SUPPORT" ] || return 0
-  local checksum_command="compute_sha2"
+  local checksum_command
+  local filename="$1"
+  local expected_checksum="$(echo "$2" | tr [A-Z] [a-z])"
 
   # If the specified filename doesn't exist, return success
-  local filename="$1"
   [ -e "$filename" ] || return 0
 
-  # If there's no expected checksum, return success
-  local expected_checksum=`echo "$2" | tr [A-Z] [a-z]`
-  [ -n "$expected_checksum" ] || return 0
+  case "${#expected_checksum}" in
+  0) return 0 ;; # empty checksum; return success
+  32) checksum_command="compute_md5" ;;
+  64) checksum_command="compute_sha2" ;;
+  *)
+    { echo
+      echo "unexpected checksum length: ${#expected_checksum} (${expected_checksum})"
+      echo "expected 0 (no checksum), 32 (MD5), or 64 (SHA2-256)"
+      echo
+    } >&2
```

Comment:
You're gonna make me cry. The caller is already redirecting to 2 to 4
On Jan 18, 2016 10:09 PM, "Mislav Marohnić" <notifications@github.com>
wrote:

> In bin/ruby-build
> <https://github.com/rbenv/ruby-build/pull/887#discussion_r50068528>:
>
> >    [ -e "$filename" ] || return 0
> >
> > -  # If there's no expected checksum, return success
> > -  local expected_checksum=`echo "$2" | tr [A-Z] [a-z]`
> > -  [ -n "$expected_checksum" ] || return 0
> > +  case "${#expected_checksum}" in
> > +  0) return 0 ;; # empty checksum; return success
> > +  32) checksum_command="compute_md5" ;;
> > +  64) checksum_command="compute_sha2" ;;
> > +  *)
> > +    { echo
> > +      echo "unexpected checksum length: ${#expected_checksum} (${expected_checksum})"
> > +      echo "expected 0 (no checksum), 32 (MD5), or 64 (SHA2-256)"
> > +      echo
> > +    } >&2
>
> If other warning messages go to fd4 right now, let's cargo-cult that and
> output to fd4 manually.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rbenv/ruby-build/pull/887/files#r50068528>.
>